File: output/CodeGen/codenet/C++/Python/s360529905.py
#!/usr/bin/python
import sys

def main():
    N = int(sys.argv[1])
    a = [int(x) for x in sys.stdin.readline().split()]
    b = sorted(a)
    frontHalf = b[N//2-1]
    backHalf = b[N//2]
    for i in range(N):
        if a[i] < backHalf:
            print(backHalf)
        else:
            print(frontHalf)

if __name__ == '__main__':
    main()

File: output/CodeGen/codenet/C++/Java/s360529905.java
public class s360529905 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        int[] b = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = in.nextInt();
            b[i] = a[i];
        }
        sort(b, b + n);
        int frontHalf = b[n / 2 - 1];
        int backHalf = b[n / 2];
        for (int i = 0; i < n; i++) {
            if (a[i] < backHalf)
                System.out.println(backHalf);
            else
                System.out.println(frontHalf);
        }
    }
}